vault/identity_store_entities.go

```diff
@@ -475,7 +490,79 @@ func (i *IdentityStore) pathEntityNameDelete() framework.OperationFunc {
 	}
 }
 
-func (i *IdentityStore) handleEntityDeleteCommon(ctx context.Context, txn *memdb.Txn, entity *identity.Entity) error {
+// pathEntityIDDelete deletes the entity for a given entity ID
+func (i *IdentityStore) handleEntityBatchDelete() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		entityIDs := d.Get("entity_ids").([]string)
+		if len(entityIDs) == 0 {
+			return logical.ErrorResponse("missing entity ids to delete"), nil
+		}
+
+		// Sort the ids by the bucket they will be deleted from
+		byBucket := make(map[string]map[string]struct{})
+		for _, id := range entityIDs {
+			bucket, ok := byBucket[i.entityPacker.BucketKey(id)]
+			if !ok {
+				bucket = make(map[string]struct{})
+			}
+
+			bucket[id] = struct{}{}
+			byBucket[i.entityPacker.BucketKey(id)] = bucket
```

Comment:
Can we more this assignment inside the `if` block above and potentially avoid repetitive assignments?

Reply:
Done